compliance/collection/nodeinventorizer/caching_inventorizer.go

```diff
@@ -0,0 +1,276 @@
+		log.Warnf("Found existing node scan backoff file - last scan may have failed. Waiting %v seconds before retrying", backoffDuration.Seconds())
+		c.backoffWaitCallback(backoffDuration)
+	}
+
+	// Write backoff duration to cache
+	backoff := inventoryWrap{BackoffDuration: c.calcNextBackoff(backoffDuration)}
+	if err := writeInventoryWrap(backoff, c.inventoryCachePath); err != nil {
+		return nil, errors.Wrap(err, " writing node scan backoff file")
+	}
+
+	// if no inventory exists, or it is too old, collect a fresh one
+	newInventory, err := collectInventory(nodeName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Write inventory to cache
+	cacheWrap := inventoryWrap{
+		ValidUntil:      time.Now().Add(c.cacheDuration),
+		BackoffDuration: 0,
+		Inventory:       newInventory,
+	}
+	if err := writeInventoryWrap(inventory, c.inventoryCachePath); err != nil {
+		return nil, errors.Wrap(err, "persisting inventory to cache")
+	}
+
+	return newInventory, nil
+}
+
+func (c *CachingScanner) calcNextBackoff(currentBackoff time.Duration) time.Duration {
+	return validateBackoff(currentBackoff * backoffMultiplier)
+}
+
+// validateBackoff ensures that a given duration does not exceed the max backoff setting
+func (c *CachingScanner) validateBackoff(backoff time.Duration) time.Duration {
+	if backoff > c.maxBackoff {
+		return c.maxBackoff
+	}
+	return backoff
+}
+
+func readInventoryWrap(path string) *inventoryWrap {
+	cacheContents, err := os.ReadFile(path)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			log.Debug("No node scan cache file found")
+			return nil
+		}
+		log.Warnf("Unable to read node scan cache. Error: %v", err)
+		return nil
+	}
+
+	// deserialize stored inventory
+	var wrap inventoryWrap
+	if err := json.Unmarshal(cacheContents, &wrap); err != nil {
+		log.Warnf("Unable to load node scan cache contents. Error: %v", err)
+		return nil
+	}
+	return &wrap
+}
+
+func writeInventoryWrap(w inventoryWrap, path string) error {
+	jsonWrap, err := json.Marshal(&w)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(path, jsonWrap, 0600)
+}
+
+// collectInventory scans the current node and returns the results as storage.NodeInventory object
+func collectInventory(nodeName string) (*storage.NodeInventory, error) {
+	metrics.ObserveScansTotal(nodeName)
+	startTime := time.Now()
+
+	log.Debug("Starting the node scan")
+
+	// uncertifiedRHEL is set to false, as scans are only supported on RHCOS for now,
+	// which only exists in certified versions
+	componentsHost, err := nodes.Analyze(nodeName, "/host/", nodes.AnalyzeOpts{UncertifiedRHEL: false, IsRHCOSRequired: true})
+
+	scanDuration := time.Since(startTime)
+	metrics.ObserveScanDuration(scanDuration, nodeName, err)
+	log.Debugf("Scanning the node took %f seconds", scanDuration.Seconds())
+
+	if err != nil {
+		log.Errorf("Error scanning node /host inventory: %v", err)
+		return nil, err
+	}
+	log.Debugf("Components found under /host: %v", componentsHost)
```

Comment:
```suggestion
	log.Debugf("Node components found from scanning: %v", componentsHost)
```
or something like this.